arithmetization/src/test/java/net/consensys/linea/zktracer/instructionprocessing/ExtCodeHashAndAccountExistenceTests.java

```diff
@@ -0,0 +1,188 @@
+            .op(EXTCODESIZE) // will return 0
+            /* next phase */
+            .push(0) // size
+            .push(0) // offset
+            .push(0) // value
+            .op(CREATE)
+            .op(DUP1)
+            .op(EXTCODEHASH) // will return KECCAK(())
+            .op(POP)
+            .op(EXTCODESIZE) // will return 0
+            .compile();
+    BytecodeRunner.of(bytecode).run();
+  }
+
+  /**
+   * Deployment happens at 0x6ff1019c622e4641f86f4bb7232b7901b8d20db6 The first EXTCODECOPY returns
+   * 0, the second one (during deployment) returns KECCAK(()) and the final one returns KECCAK(())
+   * again (no deployment occurred)
+   */
+  @Test
+  void extcodehashBeforeDuringAndAfterDeploymentDeployingEmtpyByteCode() {
+    final Bytes bytecode =
+        BytecodeCompiler.newProgram()
+            .push("6ff1019c622e4641f86f4bb7232b7901b8d20db6")
+            .op(DUP1)
+            .op(EXTCODEHASH) // will return 0
+            .op(POP)
+            .op(EXTCODESIZE) // will return 0
+            /* next phase */
+            .push("30803f3b00000000000000000000000000000000000000000000000000000000")
+            .push(0)
+            .op(MSTORE)
+            .push(4) // size
+            .push(0) // offset
+            .push(0) // value
+            .op(CREATE) // the EXTCODEHASH will return KECCAK(())
+            .op(CREATE)
+            .op(DUP1)
+            .op(EXTCODEHASH) // will return KECCAK(())
+            .op(POP)
+            .op(EXTCODESIZE) // will return 0
+            .compile();
+    BytecodeRunner.of(bytecode).run();
+  }
+
+  /**
+   * Deployment happens at 0x6ff1019c622e4641f86f4bb7232b7901b8d20db6 The first EXTCODECOPY returns
+   * 0, the second one (during deployment) returns KECCAK(()) and the final one returns KECCAK(0x00)
+   * ≡ 0xbc36789e7a1e281436464229828f817d6612f7b477d66591ff96a9e064bcc98a
+   *
+   * <p>The initialization code is ADDRESS DUP1 EXTCODEHASH EXTCODESIZE PUSH1 0x01 PUSH1 0x00 RETURN
+   */
+  @Test
+  void extcodehashBeforeDuringAndAfterDeploymentDeployingSingleZeroByte() {
+    final Bytes bytecode =
+        BytecodeCompiler.newProgram()
+            .push("6ff1019c622e4641f86f4bb7232b7901b8d20db6")
+            .op(EXTCODEHASH) // will return 0
+            .op(POP)
+            /* next phase */
+            .push("30803f3b60016000F30000000000000000000000000000000000000000000000")
+            .push(0)
+            .op(MSTORE)
+            .push(9) // size
+            .push(0) // offset
+            .push(0) // value
+            .op(CREATE) // during deployment EXTCODEHASH will return KECCAK(())
+            .op(CREATE)
+            .op(DUP1)
+            .op(EXTCODEHASH) // deployment success leaves address on the stack; will return //
+            // KECCAK(0x00)
+            .op(POP)
+            .op(EXTCODESIZE) // will return 1
+            .compile();
+    BytecodeRunner.of(bytecode).run();
+  }
+
+  @Test
+  void extcodehashOfOneself() {
+    final Bytes bytecode =
```

Comment:
Could you briefly describe the purpose of this test in a comment too?

Reply:
Done